src/sphinxcontrib/towncrier/__init__.py

```diff
@@ -118,6 +125,11 @@ def run(self) -> List[nodes.Node]:
         autoversion_mode = config.towncrier_draft_autoversion_mode
         include_empty = config.towncrier_draft_include_empty
 
+        for path in iter_towncrier_files(
+            fragments_dir=Path(config.towncrier_draft_working_directory)
+        ):
+            self.env.note_dependency(str(path))
```

Comment:
@ewjoachim doing it here seems to only work with already existing files. But if the previous run generated something, this will only add deps against those files but won't take into account the new ones. This probably needs something extra for monitoring the new files appearing in the dir because `run()` won't even get executed when that happens and the cache already exists.